tests/core/abi-utils/test_abi_utils.py

```diff
@@ -1,19 +1,55 @@
 import pytest
+import re
+from typing import (
+    Any,
+    Dict,
+    Mapping,
+    NamedTuple,
+    Sequence,
+    Tuple,
+    Type,
+    Union,
+)
+
+from eth_typing import (
+    ABI,
+    ABIComponent,
+    ABIComponentIndexed,
+    ABIConstructor,
+    ABIElement,
+    ABIError,
+    ABIEvent,
+    ABIFallback,
+    ABIFunction,
+    ABIReceive,
+    HexStr,
+)
 
 from eth_utils.abi import (
-    _abi_to_signature,
+    abi_to_signature,
+    collapse_if_tuple,
     event_abi_to_log_topic,
     event_signature_to_log_topic,
+    filter_abi_by_name,
+    filter_abi_by_type,
     function_abi_to_4byte_selector,
     function_signature_to_4byte_selector,
+    get_abi_input_names,
+    get_abi_input_types,
+    get_abi_output_names,
+    get_abi_output_types,
+    get_aligned_abi_inputs,
+    get_all_event_abis,
+    get_all_function_abis,
+    get_normalized_abi_inputs,
 )
 from eth_utils.hexadecimal import (
     encode_hex,
 )
 
-FN_ABI_A = {"name": "tokenLaunched", "type": "function", "inputs": []}
-FN_ABI_B = {"name": "CEILING", "type": "function", "inputs": []}
-FN_ABI_C = {
+FN_ABI_A: ABIFunction = {"name": "tokenLaunched", "type": "function", "inputs": []}
```

Comment:
@kclowes @fselmo @pacrob I am considering refactoring these tests. I am not satisfied with the way the contract data is parameterized and constructed in the tests. I'd rather use actual fixtures for these.

I could change these to be real fixtures and then assemble the contract in the test. The fixtures would be in a new `conftest.py` file in the `tests` dir. Currently, the `build_contract_abi` function takes element ABIs and returns the contract `ABI`. The new way would use a list of strings for the fixture names as a test parameter. When the test is invoked with the fixture names as an argument, the test will call `request.getfixturevalue(fixture_name)` for each name to get the fixture data and build the `ABI`.

I'd have some test helpers to build these easily and get reasonable type hints.
```python
def get_element_abi_from_fixture(
    fixture_name: str, request: pytest.FixtureRequest
) -> ABIElement:
    return cast(ABIElement, request.getfixturevalue(fixture_name))

def get_contract_abi_from_fixtures(
    fixture_names: Sequence[str], request: pytest.FixtureRequest
) -> ABI:
    return [get_element_abi_from_fixture(name, request) for name in fixture_names]
```

Another thought that came to mind was reusing these fixtures across libs so that I can pull them in to tests in web3.py. That would reduce duplication of these but updating them is not that difficult either, as long as we remember to do so. I'd imagine most these are never going to change anyway, new fixtures can be added later. Would sharing these between ABI test suites be a bad idea?

This effort would also involve reviewing the current tests and add cases where they are lacking.

Let me know your thoughts and if you think it'd be a good time to refactor these. Thanks!